code/game/dna/mutations/mutation_powers.dm

```diff
@@ -539,6 +548,23 @@
 		container.pixel_x = 0
 		container.pixel_y = 0
 
+/datum/spell/leap/proc/get_blocking_atom(turf/turf_to_check)
+	if(!turf_to_check)
+		return FALSE
+
+	if(turf_to_check.density)
+		return turf_to_check
+
+	for(var/mob/living/hit_mob in turf_to_check)
+		if(hit_mob.density)
+			return hit_mob
+
+	for(var/obj/hit_obj in turf_to_check)
+		if(hit_obj.density)
+			return hit_obj
```

Comment:
Why not just iterate over atoms, and then filter out within that loop? We shouldn't need to do two whole iterations here